user.py

```diff
@@ -88,9 +88,19 @@ def users(user_type):
     return render_template('users.html', users=all_users, user_type=user_type)
 
 
+def is_allowed_user(user_id):
+    user = db.session.query(User).filter_by(id=session["user_id"]).first()
+    return user.id != user_id and not user.allowed(Role.ADMIN)
```

Comment:
`user.id` should always be the same as `session["user_id"]`. You can compare them directly.